docs/content/docs/data-structures/counter.md

```diff
@@ -1,5 +1,208 @@
+
+The `SharedCounter` is a specialized [optimistic DDS]({{< relref "dds.md#optimistic-data-structures" >}}).
+It operates on communicated _deltas_ (amounts by which the shared value should be incremented or decremented), rather than direct changes to the shared value.
+In this way, it avoids the pitfalls of DDSes with simpler merge strategies, in which one user's edit may clobber another's (see [below](#why-a-specialized-dds)).
+
+Note that the `SharedCounter` only operates on integer values.
+
+### Why a specialized DDS?
+
+You may be asking yourself, why not just store the shared integer value directly in another DDS like a [SharedMap][]?
+Why incur the overhead of another runtime type?
+
+The key to the answer here is that DDSes with simpler merge strategies (like `SharedMap`) take a somewhat brute-force approach to merging concurrent edits.
+For a semantic data type like a counter, this can result in undesirable behavior.
+
+#### SharedMap Example
+
+Let's illustrate the issue with an example.
+
+Here, we will use a [SharedMap][] to store our shared integer value.
+For simplicity, it will be stored under a static key: `counter-key`.
+
+Let's say that two users are collaborating on an app with a counter widget.
+We will refer to these users as User A and User B.
+The current value of that widget is 42.
+
+What if User A and User B both simultaneously press the `+` button in their UI to increment the current value by 1?
+Behind the scenes, this increment is implemented by writing the updated value to the map.
+Each edit is then sequenced and broadcast to other collaborators.
+
+For the purpose of this example, we are ignoring the fact that two operations will never really occur at exactly the same time.
+This is intentional.
+Since increments and decrements are [commutative](https://en.wikipedia.org/wiki/Commutative_property), it shouldn't matter in what order two nearly concurrent increment operations occur; the result _should_ be the same.
+
+Here, we would expect that, after both users have processed all incoming edits, the resulting value of the counter would be 44.
+Each user pressed the `+` button once.
+42 + 1 + 1 = **44**, right?
+Again, note that we are ignoring which `+ 1` came from which user, as the order should not matter.
+
+Unfortunately, because `SharedMap` employs a _last-write-wins_ merge strategy, if User A and User B make their edits at the same time, one of the two updates will be sequnced after the other, and that value will win.
+Regardless of which user's edit was sequenced first and which was sequenced last, each user's client will update their map entry to be 42 (the current value they see) + 1 = **43**, and the latter of the two operations to be sequenced will clobber the first.
+
+So in this case, both users would see the value 43, rather than 44.
+
+This is a problem!
+If this shared counter value was being used to track store inventory, for example, that inventory would now be incorrect!
```

Comment:
This is another case where I think we're being _too_ concise. I have to stop and think about this sentence to figure out that two people are reporting newly arrived inventory (for the same type of item) in different places on a loading dock. When the reader has to figure that out, the net time-to-read the article is longer than if we use enough words to describe the situation explicitly.

Reply:
That's good feedback. Thanks! I'm starting to wonder if I may want to frame the concrete example (be it an inventory or something else) more wholistically from the start, and describe the problem from there. That is, rather than describing a more general "counter widget" case, and then trying to apply it to the inventory example after the fact.

Reply:
I completely re-worked the example. I think it is easier to follow.